demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py

```diff
@@ -491,24 +622,32 @@ def push_test_data_to_tenant(
         try:
             xsiam_client.push_to_dataset(events_test_data, rule.vendor, rule.product)
         except requests.exceptions.HTTPError:
-            logger.error(
-                (
-                    "[red]Failed pushing test data to tenant, potential reasons could be:\n - an incorrect token\n"
-                    ' - currently only http collectors configured with "Compression" as "gzip" and "Log Format" as "JSON"'
-                    " are supported, double check your collector is configured as such\n - the configured http collector "
-                    "on your tenant is disabled[/red]"
-                ),
-                extra={"markup": True},
+            system_err = (
+                f"Failed pushing test data to tenant for dataset {rule.dataset}"
             )
-            error = True
-    if error:
-        raise typer.Exit(1)
-    logger.info("[green]Test data pushed successfully[/green]", extra={"markup": True})
+            system_errors.append(system_err)
+            logger.error(f"[red]{system_err}[/red]", extra={"markup": True})
+
+    if system_errors:
+        logger.error(
+            f"[red]{FAILURE_TO_PUSH_EXPLANATION}[/red]",
+            extra={"markup": True},
+        )
+        push_test_data_test_case.system_err = "\n".join(system_errors)
+        push_test_data_test_case.result += [Failure(FAILURE_TO_PUSH_EXPLANATION)]
+    else:
+        system_out = "Test data pushed successfully"
+        push_test_data_test_case.system_out = system_out
+        logger.info(f"[green]{system_out}[/green]", extra={"markup": True})
+    push_test_data_test_case.time = (
+        datetime.now(timezone.utc) - push_test_data_test_case_start_time
+    ).total_seconds()
```

Comment:
Consider extracting this to a helper function.

Reply:
done